rdt/transformers/utils.py

```diff
@@ -427,7 +430,15 @@ def _safe_parse_datetime(value, warn=False):
 
     try:
         with warnings.catch_warnings(record=True) as captured_warnings:
-            dt = parser.parse(value)
+            try:
+                dt = parser.parse(value)
+
+            # Strings of large numbers cause parser.parse to overflow
+            except OverflowError as error:
```

Comment:
This will fix the `utc` being dropped. For some reason when there is `UTC` in the large number of values (as string) this will produce `ParserError`.
PS: you need to import it from dateutil:
```python
from dateutil.parser import UnknownTimezoneWarning, ParserError
```

```suggestion
            except (OverflowError, ParserError) as error:
```

Reply:
That's a great find!